data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -275,3 +275,109 @@ def calculate_liquidable_debt(self, protocol_name: str = None) -> list:
                 )
 
         return result_data
+
+
+class HashstackV0DBLiquidableDebtDataHandler(BaseDBLiquidableDebtDataHandler):
+    """
+    A Hashstack_V0 handler that collects data from the DB,
+    computes the liquidable debt and stores it in the database.
+
+    :cvar AVAILABLE_PROTOCOLS: A list of all available protocols.
+    """
+
+    def __init__(
+            self,
+            loan_state_class: Type[State],
+            loan_entity_class: Type[LoanEntity],
+    ):
+        super().__init__()
+        self.state_class = loan_state_class
+        self.loan_entity_class = loan_entity_class
+
+    def fetch_data(self, protocol_name: ProtocolIDs | str) -> tuple:
+        """
+        Prepares the data for the given protocol.
+        :param protocol_name: Protocol name.
+        :return: tuple
+        """
+        loan_data = self.db_connector.get_loans(
+            model=HashtackCollateralDebt,
+        )
+        interest_rate_models = self.db_connector.get_last_interest_rate_record_by_protocol_id(
+            protocol_id=protocol_name
+        )
+
+        return loan_data, interest_rate_models
+
+    def initialize_loan_entities(self, state: State, data: dict = None) -> State:
+        """
+        Initializes the loan entities in a state instance.
+        :param state: State
+        :param data: dict
+        :return: State
+        """
+
+        for instance in data:
+            loan_entity = self.loan_entity_class(
+                user=instance.user_id,
+                debt_category=instance.debt_category
```

Comment:
For `debt_category` you should use `HashtackCollateralDebt`

Reply:
What do you mean? `instance` is already a `HashtackCollateralDebt` model instance.

Reply:
It's second part of my first comment.
so here should be used `instance.debt_category` = HashtackCollateralDebt
but other fields such as `collateral` and `debt` should be from `LoanState` ordered by `last_block`